app/models/rental.rb

```diff
@@ -63,6 +64,13 @@ class Rental < ActiveRecord::Base
     event :process do
       transitions from: [:inspected, :canceled], to: :available
     end
+
+    event :process_no_show do
+      transitions from: :reserved, to: :canceled
+      after do
+        update(checked_in_at: Time.zone.now)
+      end
```

Comment:
Double check to see if this needs to be set. Technically, the customer never checked in.